src/main/webapp/app/quiz/edit/multiple-choice-question/edit-multiple-choice-question.component.ts

```diff
@@ -97,114 +71,98 @@ export class EditMultipleChoiceQuestionComponent implements OnInit, OnChanges, A
-        const lines = this.questionEditorText.split('\n').length;
-        const range = this.questionEditor.getEditor().selection.getRange();
-        this.questionEditor.getEditor().moveCursorTo(this.questionEditor.getEditor().getCursorPosition().row, Number.POSITIVE_INFINITY);
-        this.questionEditor.getEditor().insert(textToAdd);
-        range.setStart(lines, 4);
-        range.setEnd(lines, textToAdd.length - 1);
-        this.questionEditor.getEditor().selection.setRange(range, false);
+    prepareForSave(): void {
+        this.cleanupQuestion();
+        this.markdownEditor.parse();
+        this.questionUpdated.emit();
     }
 
     /**
-     * @function addHintAtCursor
-     * @desc Adds the markdown for a hint at the current cursor location
+     * @function cleanupQuestion
+     * @desc Clear the question to avoid double assignments of one attribute
      */
-    addHintAtCursor(): void {
-        this.artemisMarkdown.addHintAtCursor(this.questionEditor.getEditor());
+    private cleanupQuestion() {
+        // Reset Question Object
+        this.question.answerOptions = [];
+        this.question.text = null;
+        this.question.explanation = null;
+        this.question.hint = null;
+        this.question.hasCorrectOption = null;
     }
 
     /**
-     * This function opens the modal for the help dialog.
+     * @function domainCommandsFound
+     * @desc Get the {array} from the editor and assign its values based on the domainCommands to the corresponding question attributes one by one
+     *       1. Go trough each element of the {array} domain command
+     *       2. If the first element {string} has not second element {domainCommandIdentifier} in the array assign the string to the question text
+     *       3. else assign the first element {string} according to the second element {domainCommandIdentifier} to the corresponding attributes of the question
+     * @param {array} of markdownTextLine {string} with the corresponding domainCommand {DomainCommand} identifier
      */
-    open(content: any) {
-        this.modalService.open(content, { size: 'lg' });
-    }
+    domainCommandsFound(domainCommands: [string, DomainCommand][]): void {
+        this.cleanupQuestion();
+        let currentAnswerOption = new AnswerOption();
 
-    /**
-     * @function addExplanationAtCursor
-     * @desc Adds the markdown for an explanation at the current cursor location
-     */
-    addExplanationAtCursor(): void {
-        this.artemisMarkdown.addExplanationAtCursor(this.questionEditor.getEditor());
+        for (const command of domainCommands)  {
+
+            if (command[1] === null && command[0].length > 0) {
+                this.question.text = command[0];
+            }
+            if (command[1] instanceof CorrectOptionCommand || command[1] instanceof IncorrectOptionCommand) {
+                currentAnswerOption = new AnswerOption();
+                if (command[1] instanceof CorrectOptionCommand) {
```

Comment:
Could be simplified to currentAnswerOption.isCorrect = command[1] instanceof CorrectOptionCommand;

Reply:
Unchanged.